docs/design/keyboard-shortcuts.md

```diff
@@ -256,6 +256,69 @@ When using custom keyboard shortcuts on the web, some keyboard shortcuts that ar
+
+Use the [Office.actions.replaceShortcuts](/javascript/api/office/office.actions#replaceShortcuts) method to assign a user's custom keyboard combinations to your add-ins actions. The method takes a parameter of type `{[actionId:string]: string}`, where the `actionId`s are a subset of the action IDs that are defined in the add-in's extended manifest JSON. The values are the user's preferred key combinations. The following is an example in which some hypothetical custom key combinations are hardcoded. In an actual add-in, you create a function that gets them from the UI.
+
+```javascript
+const userCustomShortcuts = {
+    SHOWTASKPANE:"CTRL+SHIFT+1", 
+    HIDETASKPANE:"CTRL+SHIFT+2"
+};
+Office.actions.replaceShortcuts(userCustomShortcuts)
+    .then(function () {
+        console.log("Successfully registered.");
+    })
+    .catch(function (ex) {
+        if (ex.code == "ActionNotFound") {
+            console.log("ActionId does not exist.");
+        } else if (ex.code == "InvalidShortcut") {
+            console.log("Shortcut combination is invalid.");
+        }
+    });
+```
+
+To populate a UI that specifies the add-in's shortcuts that the user can customize, call the [Office.actions.getShortcuts](/javascript/api/office/office.actions#getShortcuts) method. This method returns an object of type `[actionId:string]:string|null}`, where the `actionId`s are (1) all of the action IDs that are defined in the add-in's extended manifest JSON and (2) all the customized shortcuts registered for the user in the user's roaming settings. The values are the key combinations currently assigned to the actions. The following is an example.
+
+```javascript
+Office.actions.getShortcuts()
+    .then(function (userShortcuts) {
+       for (const action in userShortcuts) {
+           let shortcut = userShortcuts[action];
+           // In a real add-in you would typically present this information in the UI.
+           console.log(action + ": " + shortcut);
+       }
+    });
+
+```
+
+As described in [Avoid key combinations in use by other add-ins](#avoid-key-combinations-in-use-by-other-add-ins), the first time a user enters a key combination that is registered to multiple actions, the user is prompted to choose which action will be associated with the combination. This can be a dilemma for a user that wants more than one of the actions to have a shortcut. You can't prevent this from happening when the conflict is between your add-in and another add-in, but you can at least prevent conflicts between the *user's customization of one of your add-in's shortcuts* and a shortcut registered by an add-in or the host Office application. To prevent a conflict, you need to warn the user about it and give them the chance to change their custom key combination.
```

Comment:
```suggestion
As described in [Avoid key combinations in use by other add-ins](#avoid-key-combinations-in-use-by-other-add-ins), the first time a user enters a key combination that is registered to multiple actions, they are prompted to choose which action will be associated with the combination. This can be a dilemma for a user that wants more than one of the actions to have a shortcut. You can't prevent this from happening when the conflict is between your add-in and another add-in, but you can at least prevent conflicts between the *user's customization of one of your add-in's shortcuts* and a shortcut registered by an add-in or the host Office application. To prevent a conflict, you need to warn the user about it and give them the chance to change their custom key combination.
```